client_http.hpp

```diff
@@ -30,6 +76,8 @@ class case_insensitive_hash {
     return seed;
   }
 };
```

Comment:
To make things simpler, we should maybe rewrite `case_insensitive_equals` and `case_insensitive_hash` to use std instead (even when using boost), and thus only have one definition of these two classes. I think `case_insensitive_hash` can use `std::hash` but have not looked into this in detail yet. Found this though after a quick search: https://stackoverflow.com/questions/2590677/how-do-i-combine-hash-values-in-c0x